### PR TITLE
Simplify get() now that we don't need it to ignorespaces

### PR DESCRIPTION
**Pull request by edemaine**
Continuation of #912 (forgot to do this change there before merging).

Open question: Should we keep the `get()` alias for `expandNextToken()`, or just use the latter?  `unget()` seems like a useful thing to keep (though it's no longer used), but we could also rename it (e.g. `pushToken`).

### CODE REVIEW

File: src/Parser.js [hunk 1]
```diff
@@ -140,17 +140,18 @@ export default class Parser {
...
+     * Switches between "text" and "math" modes.
+     * This used to reconsume nextToken in the new mode because whitespace
+     * handling differed between the two nodes; now it just updates this.mode.
      */
     switchMode(newMode) {
-        this.gullet.unget(this.nextToken);
+        //this.gullet.unget(this.nextToken);
         this.mode = newMode;
-        this.consume();
+        //this.consume();
```
Comment (kevinbarabash) [inline src/Parser.js]: remove commented out code

File: src/Parser.js [hunk 2]
```diff
@@ -140,17 +140,18 @@ export default class Parser {
...
-        this.nextToken = this.gullet.get(false);
+        this.nextToken = this.gullet.get();
     }
 
     /**
-     * Switches between "text" and "math" modes, reconsuming nextToken
-     * in case it would be read differently in the new mode.
+     * Switches between "text" and "math" modes.
+     * This used to reconsume nextToken in the new mode because whitespace
+     * handling differed between the two nodes; now it just updates this.mode.
```
Comment (kevinbarabash) [inline src/Parser.js]: Describing the previous behavior isn't necessary.

File: src/Parser.js (hunk 1 above)
Comment (kevinbarabash) [reply src/Parser.js]: I'm curious though why the `unget` and `consume` are no longer necessary.

Comment (edemaine) [reply src/Parser.js]: It used to be that `get` got passed true or false according to whether math mode was on, i.e., it ate spaces if it was in math mode.  But that's no longer the case (having moved that into parser, where it belongs according to TeX behavior), so now get doesn't depend on whether math mode is on, so there's no effect to ungetting and regetting.

**Comment by edemaine** [comment]
@kevinbarabash I've removed the excess documentation and commented-out old behavior.

**Comment by kevinbarabash** [comment]
I'll finish reviewing this evening.

File: src/MacroExpander.js [hunk 3]
```diff
@@ -227,36 +225,17 @@ export default class MacroExpander implements MacroContextInterface {
...
     }
 
     /**
      * Undo the effect of the preceding call to the get method.
-     * A call to this method MUST be immediately preceded and immediately followed
-     * by a call to get.  Only used during mode switching, i.e. after one token
-     * was got in the old mode but should get got again in a new mode
-     * with possibly different whitespace handling.
      */
     unget(token: Token) {
```
Comment (kevinbarabash) [inline src/MacroExpander.js]: Since it's not being used and it's easy enough to do `this.stack.push(token)`, let's get rid of this.

Comment (edemaine) [reply src/MacroExpander.js]: I think it's still a nice interface to keep, in case we want to use it in macros.js. Though still happy to rename it...

Comment (kevinbarabash) [reply src/MacroExpander.js]: If you rename it, please change all `this.stack.push(token)`s to use it.

**Comment by edemaine** [comment]
I dropped the `get()` alias (opting for `expandNextToken` instead), renamed `unget()` to `pushToken()`, and used it in two places as suggested by @kevinbarabash (good catch!).  There is one exception: I couldn't easily change `this.stack.push(...expansion)` given the different argument type.  Should I change `pushToken()` to use `arguments`...?  There's also a new doc string for `pushToken` that'd be worth reading.

**Comment by kevinbarabash** [comment]
I think it's fine for `pushToken` to only accept a single token for now.  If we find we're needing to push multiple tokens at once in more places in the future we can revisit at that time.

**Comment by kevinbarabash** [comment]
@edemaine thanks the PR.

**Pull request merged**